File: java/estate.java
import java.util.Scanner;

public class estate {
	static final int[] options = new int[1000000];

	static {
		int count = 1;
		for (int i = 2; i < options.length; i += count + 1) {
			for (int j = i; j < options.length; j += count)
				options[j - 1]++;
			count++;
		}
	}

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		int input;
		while ((input = scan.nextInt()) != 0)
			System.out.println(options[input - 1]);
		scan.close();
	}
}

File: ts/src/estate.ts
import readline = require('readline')

function generate (): void {
  let count = 1
  for (let i = 2; i < options.length; i += ++count + 1) {
    for (let j = i; j < options.length; j += count) {
      options[j - 1]++
    }
  }
}

const options: number[] = Array(1000000).fill(0)

generate()

readline
  .createInterface({
    input: process.stdin,
    output: process.stdout
  })
  .on('line', (line: string) => {
    if (line === '0') {
      process.exit(0)
    }
    console.log(options[parseInt(line) - 1])
  })
  .on('close', () => {
    process.exit(0)
  })
